src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private static final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Movie " + movie + "saving to DB failed " + e);
```

Comment:
```suggestion
            throw new DataProcessingException("Movie " + movie + "saving to DB failed ", e);
```
В ексепшинах ти ще так можеш cause передавати (але це звичайно ще конструкторі треба додати (message, Throwable cause). Так на вищих рівнях ти зможеш діставати причину ексепшена через e.getCause(), що може бути зручно при логуванні і дебазі